src/data/dfinsupp/basic.lean

```diff
@@ -1929,3 +1935,31 @@ add_monoid_hom.congr_fun (comp_lift_add_hom h.to_add_monoid_hom g) f
 end add_equiv
 
 end
+
+section finite_infinite
+
+instance dfinsupp.fintype {ι : Sort*} {π : ι → Sort*} [decidable_eq ι] [Π i, has_zero (π i)]
+  [fintype ι] [∀ i, fintype (π i)] :
+  fintype (Π₀ i, π i) :=
+fintype.of_equiv (Π i, π i) dfinsupp.equiv_fun_on_fintype.symm
+
+/-- Weaker form of `dfinsupp.fintype`. -/
+instance dfinsupp.finite {ι : Sort*} {π : ι → Sort*} [Π i, has_zero (π i)]
+  [finite ι] [∀ i, finite (π i)] :
+  finite (Π₀ i, π i) :=
```

Comment:
Thanks for adding this, but the finsupp version is missing right? [pi.finite](https://leanprover-community.github.io/mathlib_docs/data/finite/basic.html#pi.finite) does exist.

Reply:
> And should we also add `dfinsupp.infinite_of_right` for completeness?

This was already in the first commit, did you mean something else?

> [pi.finite](https://leanprover-community.github.io/mathlib_docs/data/finite/basic.html#pi.finite) does exist, but `function.finite` and `function.fintype` are also missing: maybe that's indication that the `pi` versions are enough and `function.infinite_of_left/right` are actually not necessary?

I don't think it's strictly necessary to add them, especially since Lean 4 fixes the dependent instance bug that would require these duplications. So we could just hope that nobody triggers the bug in this context before we migrate to Lean 4. But now the instances are already here, I don't see an objection to keeping them because that would be much less work than adding them again when the dependent instance bug shows up.

Reply:
Sorry 😅 It seems what's missing is:
```
lemma dfinsupp.infinite_of_exists_right {ι : Type*} {π : ι → Type*} 
  (i : ι) [infinite $ π i] [Π i, has_zero (π i)] :
  infinite (Π₀ i : ι, π i) :=
```